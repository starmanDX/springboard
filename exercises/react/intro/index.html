<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Part1</title>
</head>
<body>
    <div id="root"></div>
    <script crossorigin src="https://unpkg.com/react@16/umd/react.development.js"></script>
    <script crossorigin src="https://unpkg.com/react-dom@16/umd/react-dom.development.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/babel-standalone/6.26.0/babel.js"></script>
    <script src="FirstComponent.jsx" type="text/jsx"></script>
    <script src="NamedComponent.jsx" type="text/jsx"></script>
    <script src="App.jsx" type="text/jsx"></script>
</body>
</html>