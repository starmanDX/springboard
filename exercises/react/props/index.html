<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>JSX Part2</title>
</head>
<body>
    <h1>JSX Props Part 2 - Tweets</h1>
    <div id="root"></div>
    <script crossorigin src="https://unpkg.com/react@16/umd/react.development.js"></script>
    <script crossorigin src="https://unpkg.com/react-dom@16/umd/react-dom.development.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/babel-standalone/6.26.0/babel.js"></script>
    <script src="Tweet.jsx" type="text/jsx"></script>
    <script src="App.jsx" type="text/jsx"></script>
</body>
</html>