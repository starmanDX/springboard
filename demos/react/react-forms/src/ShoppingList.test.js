import React from "react";
import { render, fireEvent } from "@testing-library/react";
import ShoppingList from "./ShoppingList";

it("should render without crashing", () => {
  render(<ShoppingList />);
});

it("should match the snapshot", () => {
  const { asFragment } = render(<ShoppingList />);
  expect(asFragment()).toMatchSnapshot();
});

it("should add a new item", () => {
  const { queryByText, getByLabelText } = render(<ShoppingList />);
  const input = getByLabelText("Product");
  const btn = queryByText("Add Item");
  expect(queryByText(`Product Name: Chocolate Milk`)).not.toBeInTheDocument();
  fireEvent.change(input, { target: { value: "Chocolate Milk" } });
  fireEvent.click(btn)
  expect(queryByText(`Product Name: Chocolate Milk`)).toBeInTheDocument();
});
