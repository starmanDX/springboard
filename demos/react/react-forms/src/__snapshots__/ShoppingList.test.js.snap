// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <h3>
    Shopping List
  </h3>
  <form>
    <label
      for="name"
    >
      Product 
    </label>
    <input
      id="name"
      name="name"
      placeholder="Item Name"
      type="text"
      value=""
    />
    <label
      for="qty"
    >
      Quantity: 1 
    </label>
    <input
      id="qty"
      max="10"
      min="1"
      name="qty"
      type="range"
      value="1"
    />
    <button>
      Add Item
    </button>
  </form>
  <div>
    <ul>
      <li>
        Product Name: Peanut Butter
      </li>
      <li>
        Quantity: 2
      </li>
    </ul>
    <ul>
      <li>
        Product Name: Jelly
      </li>
      <li>
        Quantity: 1
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
